src/helpers/courses.js

```diff
@@ -32,24 +32,32 @@ exports.list = async (query) => {
   return [
     ...intraCourse,
     ...interCourse.filter(course => course.companies.includes(query.company))
-      .map((course) => ({
-          ...omit(course, ['companies']),
-          trainees: course.trainees.filter(t => query.company === t.company._id.toHexString()),
-        })),
+      .map(course => ({
+        ...omit(course, ['companies']),
+        trainees: course.trainees.filter(t => query.company === t.company._id.toHexString()),
+      })),
   ];
 };
 
-exports.getCourse = async courseId => Course.findOne({ _id: courseId })
-  .populate({ path: 'company', select: 'name' })
-  .populate({ path: 'program', select: 'name learningGoals' })
-  .populate('slots')
-  .populate({
-    path: 'trainees',
-    select: 'identity.firstname identity.lastname local.email company contact ',
-    populate: { path: 'company', select: 'name' },
-  })
-  .populate({ path: 'trainer', select: 'identity.firstname identity.lastname' })
-  .lean();
+exports.getCourse = async (courseId, loggedUser) => {
+  const userHasVendorRole = !!get(loggedUser, 'role.vendor');
+  const userCompanyId = get(loggedUser, 'company._id') || null;
+  // If user has vendor role, all trainees are populated. Otherwise, only trainees from user's company are.
```

Comment:
en fait avec ce commentaire tu expliques ce que tu fais - ce qu'on lit avec le code. La commentaire est utile si il donne une information en plus, typiquement ici pourquoi on ne popule pas sans role vendeur la logique metier derriere

tu peux mettre : "A client coach user is not supposed to read trainees from other companies info - espacially for INTER_B2B courses."